lib/containers/common.pm

```diff
@@ -25,13 +26,14 @@ sub is_unreleased_sle {
 }
 
 sub activate_containers_module {
-    my ($running_version, $sp, $host_distri) = get_os_release;
-    my $suseconnect = script_output_retry('SUSEConnect --status-text', timeout => 240, retry => 3, delay => 60);
-    if ($suseconnect !~ m/Containers/) {
-        $running_version eq '12' ? add_suseconnect_product("sle-module-containers", 12) : add_suseconnect_product("sle-module-containers");
-        $suseconnect = script_output_retry('SUSEConnect --status-text', timeout => 240, retry => 3, delay => 60);
+    my $running_version = shift;
+    my $json = Mojo::JSON::decode_json(script_output_retry('SUSEConnect -s', timeout => 240, retry => 3, delay => 60));
+    my $registered = 0;    # We expect that containers module is not registered
+    foreach (@$json) {
+        $registered = 1 if ($_->{identifier} =~ 'container' && $_->{status} =~ 'Registered');
     }
-    record_info('SUSEConnect', $suseconnect);
+    ($registered && $running_version eq '12') ? add_suseconnect_product("sle-module-containers", 12) : add_suseconnect_product("sle-module-containers");
```

Comment:
Shouldn't it be
```suggestion
    (!($registered) && $running_version eq '12') ? add_suseconnect_product("sle-module-containers", 12) : add_suseconnect_product("sle-module-containers");
```
In words: We add the container product only, if it is not yet registered, right?

Reply:
I changed the variable name to `$not_registered`.